### PR TITLE
Add ListObjects V2 API

### LINKED ISSUE #460: api: Implement support for ListObjectsV2. supporting new S3 list version 2 type.

### PR DESCRIPTION
**Pull request by vadmeste**
Fixes #460

### CODE REVIEW

File: src/test/unit/test.js [hunk 1]
```diff
@@ -1091,9 +1091,48 @@ describe('Client', function() {
       })
     })
```
Comment (krishnasrinivas) [inline src/test/unit/test.js]: can you write the test cases in functional tests? we are moving away from unit tests to functional tests for API testing

`minio-js/src/test/functional/functional-tests.js`

File: docs/API.md [hunk 2]
```diff
@@ -391,6 +392,68 @@ stream.on('error', function(err) { console.log(err) } )
...
+| ---- | ---- | ---- |
+| `bucketName` | _string_ | Name of the bucket. |
+| `prefix`  | _string_  |  The prefix of the objects that should be listed (optional, default `''`). |
+| `recursive`  | _bool_  | `true` indicates recursive style listing and `false` indicates directory style listing delimited by '/'. (optional, default `false`).  |
+
+
+__Return Value__
+
+
+<table>
```
Comment (harshavardhana) [inline docs/API.md]: Remove the HTML code, we are not using this style anymore. Refer to other APIs.

File: src/main/minio.js [hunk 3]
```diff
@@ -1148,6 +1148,120 @@ export class Client {
...
+    if (maxKeys) {
+      if (maxKeys >= 1000) {
+        maxKeys = 1000
+      }
+      queries.push(`max-keys=${maxKeys}`)
+    }
+    queries.sort()
+    var query = ''
+    if (queries.length > 0) {
+      query = `${queries.join('&')}`
```
Comment (brendanashworth) [inline src/main/minio.js]: You could use [`querystring#stringify()`](https://nodejs.org/api/querystring.html#querystring_querystring_stringify_obj_sep_eq_options) here if you'd like.

File: src/main/minio.js [hunk 4]
```diff
@@ -1148,6 +1148,120 @@ export class Client {
...
+  // * `stream` _Stream_: stream emitting the objects in the bucket, the object is of the format:
+  //   * `stat.key` _string_: name of the object
+  //   * `stat.size` _number_: size of the object
+  //   * `stat.etag` _string_: etag of the object
+  //   * `stat.lastModified` _string_: modified time stamp
+  listObjectsV2(bucketName, prefix, recursive) {
+    if (prefix === undefined) prefix = ''
+    if (recursive === undefined) recursive = false
+    if (!isValidBucketName(bucketName)) {
+      throw new errors.InvalidBucketNameError('Invalid bucket name: ' + bucketName)
```
Comment (brendanashworth) [inline src/main/minio.js]: You could use a template string here: `'Invalid bucket name: ${bucketName}'`

Comment (krishnasrinivas) [inline src/main/minio.js]: it has to be a backtick, not quote

File: src/main/minio.js (hunk 3 above)
Comment (krishnasrinivas) [inline src/main/minio.js]: all our apis should use querystring module, better to make the changes as a separate patch after making sure that it does not cause problems for signature calculation.

File: src/main/minio.js (hunk 4 above)
Comment (brendanashworth) [inline src/main/minio.js]: markdown 😉
